### ISSUE TITLE
"flatfs: too many open files" error from (Darwin) ipget v0.3.1 when downloading a large file

### ISSUE THREAD

**Issue by laser** [ipfs/go-ipfs#6588]
ipget v.0.3.1 **occasionally** produces an error "flatfs: too many open files" in **some OSX environments** when downloading a piece of content.

To reproduce (on OSX):

```shell
cd $(mktemp -d) \
  && curl 'https://dist.ipfs.io/ipget/v0.3.1/ipget_v0.3.1_darwin-amd64.tar.gz' > ipget_v0.3.1_darwin-amd64.tar.gz \
  && tar -xzf /tmp/ipget_v0.3.1_darwin-amd64.tar.gz \
  && cd ipget \
  && ./ipget -o /tmp/stuff QmRDcUxfpPY9a1vR3T4vRgxHtWHyy9m3xuMQtj8P749r4e
```

**Comment by hannahhoward** [comment]
for reference, I received this an an MDNS error while downloading, thought it eventually succeeded...

![Screen Shot 2019-08-19 at 4 36 01 PM](https://user-images.githubusercontent.com/1450680/63306501-ab3cfc00-c29f-11e9-9228-c16419cf72fc.png)

**Comment by hannahhoward** [comment]
FWIW, I don't think any of these are preventing the download, which to me feels like they ought to be WARNING level logs, which I assume ipget doesn't output @Stebalien

**Comment by hannahhoward** [comment]
also, ipget 0.3.1 is.... using some version of IPFS which is still on GX... so I suspect it's a wee bit out of date.

**Comment by hannahhoward** [comment]
What is the status on: https://github.com/ipfs/ipget/pull/59

**Comment by Stebalien** [comment]
I made a PR friday (https://github.com/ipfs/ipget/pull/59) but need to figure out why CI is failing.

**Comment by hannahhoward** [comment]
I'll look at it @Stebalien to save you some trouble

**Comment by Stebalien** [comment]
Thanks!

**Comment by Stebalien** [comment]
Could someone try out 0.4.0 from https://ipfs.io/ipfs/QmVE4kPH5reJRvazaZwZK3Rrp95nGxkKtr253ZF6FJxVv9/?

**Comment by Stebalien** [comment]
Ipget 0.4.0 has been released. Closing as 0.3.1 uses an ancient version of go-ipfs.

**Comment by momack2** [comment]
Can we validate this issue was resolved for 0.4.0?
